OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxFileActions.java

```diff
@@ -0,0 +1,115 @@
+package latexstudio.editor;
+
+import com.dropbox.core.DbxClient;
+import com.dropbox.core.DbxEntry;
+import com.dropbox.core.DbxException;
+import com.dropbox.core.DbxWriteMode;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.swing.DefaultListSelectionModel;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import latexstudio.editor.files.FileService;
+import latexstudio.editor.remote.DbxEntryDto;
+import latexstudio.editor.remote.DbxState;
+import latexstudio.editor.remote.DbxUtil;
+import latexstudio.editor.util.ApplicationUtils;
+import org.apache.commons.io.IOUtils;
+import org.openide.util.Exceptions;
+
+public class DbxFileActions {
+
+    private EditorTopComponent etc;
+
+    public DbxFileActions(EditorTopComponent etc) {
+        this.etc = etc;
+    }
+
+    public void saveProgress(DropboxRevisionsTopComponent drtc) {
+        DbxClient client = DbxUtil.getDbxClient();
+
+        if (client == null) {
+            return;
+        }
+
+        String sourceFileName = ApplicationUtils.getTempSourceFile();
+        File file = new File(sourceFileName);
+
+        FileInputStream inputStream = null;
+        try {
+            inputStream = new FileInputStream(file);
+        } catch (FileNotFoundException ex) {
+            Exceptions.printStackTrace(ex);
+        }
```

Comment:
I'm also wondering why this single line has got a separate try-catch block, instead of being placed with the rest of the code that uses the stream. I know this is something that has only been moved to a new class, but perhaps it's worth refactoring.